src/map/map.c

```diff
@@ -2268,46 +2268,45 @@ static struct map_session_data *map_charid2sd(int charid)
  * (without sensitive case if necessary)
  * return map_session_data pointer or NULL
  *------------------------------------------*/
-static struct map_session_data *map_nick2sd(const char *nick)
+static struct map_session_data *map_nick2sd(const char *nick, bool allow_partial)
 {
 	struct map_session_data* sd;
 	struct map_session_data* found_sd;
 	struct s_mapiterator* iter;
 	size_t nicklen;
 	int qty = 0;
 
-	if( nick == NULL )
+	if (nick == NULL)
 		return NULL;
 
 	nicklen = strlen(nick);
 	iter = mapit_getallusers();
 
 	found_sd = NULL;
 	for (sd = BL_UCAST(BL_PC, mapit->first(iter)); mapit->exists(iter); sd = BL_UCAST(BL_PC, mapit->next(iter))) {
-		if( battle_config.partial_name_scan )
-		{// partial name search
-			if( strnicmp(sd->status.name, nick, nicklen) == 0 )
-			{
+		if (battle_config.partial_name_scan && allow_partial) {
```

Comment:
i not sure, but probably if better move outside of ``for`` for better performance

Reply:
Done, thank you